source/portable/NetworkInterface/RM46L852/NetworkInterface.c

```diff
@@ -0,0 +1,369 @@
+/*
+ * FreeRTOS+TCP <DEVELOPMENT BRANCH>
+ * Copyright (C) 2023 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * https://www.FreeRTOS.org
+ * https://github.com/FreeRTOS
+ */
+
+/**
+ * @file NetworkInterface.c
+ * @brief Implements the Network Interface driver for the TI Hercules RM46 board.
+ */
+
+#include "FreeRTOS.h"
+#include "list.h"
+#include "queue.h"
+#include "semphr.h"
+#include "task.h"
+
+/* FreeRTOS+TCP includes. */
+#include "FreeRTOS_IP.h"
+#include "FreeRTOS_Sockets.h"
+#include "FreeRTOS_IP_Private.h"
+#include "NetworkBufferManagement.h"
+#include "NetworkInterface.h"
+
+#include "emac.h"
+#include "mdio.h"
+#include "phy_dp83640.h"
+#include "sci.h"
+
+
+#if ipconfigETHERNET_DRIVER_FILTERS_FRAME_TYPES == 0
+    #define ipCONSIDER_FRAME_FOR_PROCESSING( pucEthernetBuffer )    eConsiderFrameForProcessing( ( pucEthernetBuffer ) )
+#else
+    #define ipCONSIDER_FRAME_FOR_PROCESSING( pucEthernetBuffer )    eProcessBuffer
+#endif
+
+
+TaskHandle_t receiveTaskHandle = NULL;
+
+void prvEMACHandlerTask( void * pvParameters  );
+void prvProcessFrame( NetworkBufferDescriptor_t * pxBufferDescriptor );
+uint8   emacAddress[6U] =   {0x00U, 0x08U, 0xEEU, 0x03U, 0xA6U, 0x6CU};
+
+/*
+ * This function is used to call EMAC API to perform low level hardware
+ * initializations.
+ * Also, creates prvEMACHandlerTask for handling received packets.
+ */
+
+BaseType_t xNetworkInterfaceInitialise( void )
+{
+    BaseType_t xFirstCall = pdTRUE;
+    BaseType_t xTaskCreated;
+
+    if (EMACHWInit(emacAddress) == EMAC_ERR_OK)
+    {
+        if( ( xFirstCall == pdTRUE ) || ( receiveTaskHandle == NULL ) )
```

Comment:
You create and set `xFirstCall` inside of this function. This check will always return true.